### ISSUE TITLE
Development and Master fail

### ISSUE THREAD

**Issue by shukon** [automl/SMAC3#312]
```
======================================================================
FAIL: adding some rundata to RunHistory2EPM4Cost and impute censored data
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/automl/SMAC3/test/test_runhistory/test_runhistory2epm.py", line 207, in test_cost_with_imputation
    decimal=3)
  File "/home/travis/miniconda/envs/testenv/lib/python3.5/site-packages/numpy/testing/utils.py", line 962, in assert_array_almost_equal
    precision=decimal)
  File "/home/travis/miniconda/envs/testenv/lib/python3.5/site-packages/numpy/testing/utils.py", line 778, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not almost equal to 3 decimals
(mismatch 33.33333333333333%)
 x: array([[   1.   ],
       [  16.451],
       [ 200.   ]])
 y: array([[   1.   ],
       [  16.422],
       [ 200.   ]])
-------------------- >> begin captured logging << --------------------
smac.scenario.scenario.Scenario: INFO: Output to smac3-output_2017-08-30_13:23:59_(384673)
smac.scenario.scenario.Scenario: DEBUG: Output directory does not exist! Will be created.
smac.scenario.scenario.Scenario: DEBUG: Writing scenario-file to smac3-output_2017-08-30_13:23:59_(384673)_run1/scenario.txt.
smac.scenario.scenario.Scenario: DEBUG: Scenario Options:
smac.scenario.scenario.Scenario: DEBUG: cost_for_crash = 2147483647.0
smac.scenario.scenario.Scenario: DEBUG: deterministic = False
smac.scenario.scenario.Scenario: DEBUG: run_obj = runtime
smac.scenario.scenario.Scenario: DEBUG: intensification_percentage = 0.5
smac.scenario.scenario.Scenario: DEBUG: cutoff = 20.0
smac.scenario.scenario.Scenario: DEBUG: shared_model = False
smac.scenario.scenario.Scenario: DEBUG: maxR = 2000
smac.scenario.scenario.Scenario: DEBUG: output_dir = smac3-output_2017-08-30_13:23:59_(384673)
smac.scenario.scenario.Scenario: DEBUG: minR = 1
smac.scenario.scenario.Scenario: DEBUG: execdir = .
smac.scenario.scenario.Scenario: DEBUG: overall_obj = par10
smac.scenario.scenario.Scenario: DEBUG: algo_runs_timelimit = inf
smac.scenario.scenario.Scenario: DEBUG: initial_incumbent = DEFAULT
smac.scenario.scenario.Scenario: DEBUG: wallclock_limit = inf
smac.scenario.scenario.Scenario: DEBUG: abort_on_first_run_crash = True
smac.scenario.scenario.Scenario: DEBUG: always_race_default = False
smac.scenario.scenario.Scenario: DEBUG: ta_run_limit = inf
smac.scenario.scenario.Scenario: DEBUG: Deactivate output directory.
smac.scenario.scenario.Scenario: INFO: No output directory for scenario logging specified -- scenario will not be logged.
smac.scenario.scenario.Scenario: DEBUG: Scenario Options:
smac.scenario.scenario.Scenario: DEBUG: cost_for_crash = 2147483647.0
smac.scenario.scenario.Scenario: DEBUG: deterministic = False
smac.scenario.scenario.Scenario: DEBUG: run_obj = runtime
smac.scenario.scenario.Scenario: DEBUG: intensification_percentage = 0.5
smac.scenario.scenario.Scenario: DEBUG: cutoff = 20.0
smac.scenario.scenario.Scenario: DEBUG: shared_model = False
smac.scenario.scenario.Scenario: DEBUG: maxR = 2000
smac.scenario.scenario.Scenario: DEBUG: output_dir = 
smac.scenario.scenario.Scenario: DEBUG: minR = 1
smac.scenario.scenario.Scenario: DEBUG: execdir = .
smac.scenario.scenario.Scenario: DEBUG: overall_obj = par10
smac.scenario.scenario.Scenario: DEBUG: algo_runs_timelimit = inf
smac.scenario.scenario.Scenario: DEBUG: initial_incumbent = DEFAULT
smac.scenario.scenario.Scenario: DEBUG: wallclock_limit = inf
smac.scenario.scenario.Scenario: DEBUG: abort_on_first_run_crash = True
smac.scenario.scenario.Scenario: DEBUG: always_race_default = False
smac.scenario.scenario.Scenario: DEBUG: ta_run_limit = inf
smac.runhistory.runhistory2epm.RunHistory2EPM4Cost: DEBUG: Transform runhistory into X,y format
smac.runhistory.runhistory2epm.RunHistory2EPM4Cost: DEBUG: No censored data found, skip imputation
smac.runhistory.runhistory2epm.RunHistory2EPM4Cost: DEBUG: Converted 1 observations
smac.runhistory.runhistory2epm.RunHistory2EPM4Cost: DEBUG: Transform runhistory into X,y format
smac.runhistory.runhistory2epm.RunHistory2EPM4Cost: DEBUG: No censored data found, skip imputation
smac.runhistory.runhistory2epm.RunHistory2EPM4Cost: DEBUG: Converted 2 observations
smac.runhistory.runhistory2epm.RunHistory2EPM4Cost: DEBUG: Transform runhistory into X,y format
smac.runhistory.runhistory2epm.RunHistory2EPM4Cost: DEBUG: 1 TIMOUTS, 2 censored, 1 regular
smac.epm.rfr_imputator.RFRImputator: DEBUG: Going to impute 2 y-values with <smac.epm.rf_with_instances.RandomForestWithInstances object at 0x7fd64065e588>
smac.epm.rfr_imputator.RFRImputator: DEBUG: Iteration 1 of 2
smac.epm.rfr_imputator.RFRImputator: DEBUG: Going to replace 2 nan-value(s) with max(captime, predicted mean)
smac.epm.rfr_imputator.RFRImputator: DEBUG: Change: 0.000000
smac.epm.rfr_imputator.RFRImputator: DEBUG: Iteration 2 of 2
smac.epm.rfr_imputator.RFRImputator: DEBUG: Change: 0.417230
smac.epm.rfr_imputator.RFRImputator: DEBUG: Imputation used 2/2 iterations, last_change=0.417230
smac.runhistory.runhistory2epm.RunHistory2EPM4Cost: DEBUG: Converted 3 observations
--------------------- >> end captured logging << ---------------------
```

**Comment by mlindauer** [comment]
@sfalkner found a critical bug in the pyrfr and fixed it in the newest pyrfr version. My guess would be that the bugfix also affects our unit tests (since travis-ci uses always the newest version). So most likely the code is not broken but the unit test.

**Comment by mlindauer** [comment]
@shukon Thank you very much for reporting this issue :-)

**Comment by sfalkner** [comment]
The latest version will influence any predictions (even with fixed seed), so it's probably the unittest.
I don't know where the reference value come from, but SMAC actually doesn't have to unittest the predictions of the forest itself. That is already done in the pyrfr unittests.

**Comment by shukon** [comment]
fixed in #314
